jax/_src/numpy/ufuncs.py

```diff
@@ -159,8 +167,8 @@ def _arccosh(x: ArrayLike, /) -> Array:
 invert = _one_to_one_unop(np.invert, lax.bitwise_not)
 negative = _one_to_one_unop(np.negative, lax.neg)
 positive = _one_to_one_unop(np.positive, lambda x: lax.asarray(x))
-floor = _one_to_one_unop(np.floor, lax.floor, True)
-ceil = _one_to_one_unop(np.ceil, lax.ceil, True)
+floor = _one_to_one_unop(np.floor, _preserve_int_dtype_unop(lax.floor), False)
+ceil = _one_to_one_unop(np.ceil, _preserve_int_dtype_unop(lax.ceil), False)
```

Comment:
At this point it's no longer really a one-to-one operation, so it would be better to define the function than add another level of indirection when defining it automatically.